server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGates.java

```diff
@@ -263,17 +267,31 @@ public boolean currentUserHasWritePermission() {
     return hasWritePermission;
   }
 
-  private void validateCondition(Metric metric, String operator, @Nullable String warningThreshold, @Nullable String errorThreshold, @Nullable Integer period) {
+  private void validateCondition(long qGateId, Metric metric, String operator, @Nullable String warningThreshold, @Nullable String errorThreshold, @Nullable Integer period) {
     Errors errors = new Errors();
     validateMetric(metric, errors);
     checkOperator(metric, operator, errors);
     checkThresholds(warningThreshold, errorThreshold, errors);
     checkPeriod(metric, period, errors);
+    checkConditionDoesNotAlreadyExistOnSameMetricAndPeriod(qGateId, metric, period, errors);
     if (!errors.isEmpty()) {
       throw new BadRequestException(errors);
     }
   }
 
+  private void checkConditionDoesNotAlreadyExistOnSameMetricAndPeriod(long qGateId, Metric metric, @Nullable final Integer period, Errors errors) {
+    Collection<QualityGateConditionDto> conditions = conditionDao.selectForQualityGate(qGateId);
+    if (conditions.isEmpty()) {
+      return;
+    }
+    boolean conditionExists = from(conditions).anyMatch(new MatchQualityGate(metric.getId(), period));
+    if (period == null) {
```

Comment:
I would use the ternary operator to define the error message and then have only one error check. Something similar to:
```
String errorMessage = period==null
  ? format("Condition on metric '%s' already exists.", metric.getName())
  : format("Condition on metric '%s' over leak period already exists.", metric.getName());
errors.check(!conditionExists, errorMessage);
```